### PR TITLE
Changes to enable manylinux2_28 wheels for ROCm

### PR DESCRIPTION
**Pull request by jithunnair-amd**
* Add manylinux2_28 case to `manywheel/build_docker.sh`
* Update steps to install libraries such as magma for ROCm
* Clean up old-ROCm-version code
* Use _GLIBCXX_USE_CXX11_ABI=1: https://github.com/ROCm/builder/pull/51/commits/9daa4e4b540b7c39c82b8d60b45d93b866dd7241
* Install numpy<2.0.0 for PyTorch versions<2.4: https://github.com/ROCm/builder/pull/51/commits/6bf4bfd163f3810f6e2d016145fde94dca688527
* Add conditions for ROCm6.2 and ROCm6.3 in `common/install_miopen.sh`
* Upgrade CMake version to 3.28.4 (need minimum of 3.21 to get `enable_langugage(HIP)` support, but 3.21 has a bug that inserts dynamic linker flags `-Wl,no-as-needed` in static linker flags causing build failure)
* Skip unnecessary installation of epel rpm from expired link for CentOS7
* Install epel-release-7 from cached rpm to enable ius-release to be installed for Almalinux8

TODO:
- [x] Cherry-pick changes to rocm6.2 branch - Not needed since ROCm6.2.x stays on CentOS7-based flow

Tested via: http://ml-ci-internal.amd.com:8080/job/pytorch/job/dev/job/manylinux_rocm_wheels_test/158/

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
For "Install numpy<2.0.0 for PyTorch versions<2.4: https://github.com/ROCm/builder/commit/6bf4bfd163f3810f6e2d016145fde94dca688527":
```
[2024-07-17T20:18:01.140Z] +++ printf %3d%03d%03d%03d 2 4
[2024-07-17T20:18:01.140Z] ++ [[   1013001000 -ge   2004000000 ]]
[2024-07-17T20:18:01.140Z] ++ retry pip install -q 'numpy<2.0.0'
[2024-07-17T20:18:01.140Z] ++ pip install -q 'numpy<2.0.0'
```

**Comment by jithunnair-amd** [comment]
For "Use _GLIBCXX_USE_CXX11_ABI=1: https://github.com/ROCm/builder/commit/9daa4e4b540b7c39c82b8d60b45d93b866dd7241"
```
[2024-07-17T20:18:04.647Z] [0m++ [[ '' == *\c\x\x\1\1\-\a\b\i* ]]
[2024-07-17T20:18:04.647Z] ++ [[ rocm6.1.0 == *\r\o\c\m* ]]
[2024-07-17T20:18:04.647Z] ++ export _GLIBCXX_USE_CXX11_ABI=1
[2024-07-17T20:18:04.647Z] ++ _GLIBCXX_USE_CXX11_ABI=1
```

File: manywheel/build_docker.sh [hunk 1]
```diff
@@ -76,21 +76,22 @@ case ${GPU_ARCH_TYPE} in
         MANY_LINUX_VERSION="aarch64"
         DOCKERFILE_SUFFIX="_cuda_aarch64"
         ;;
-    rocm)
+    rocm|rocm-manylinux_2_28)
```
Comment (pruthvistony) [inline manywheel/build_docker.sh:79]: Better to keep this as a separate case, following the same pattern.

**Review by pruthvistony — APPROVED** [review]

File: manywheel/build_docker.sh (hunk 1 above)
Comment (jithunnair-amd) [reply manywheel/build_docker.sh:79]: Actually, that's what I had started out with, following the pattern for cuda in the file, but then I came across this nifty way to cover more than one case in bash switch statement. And given the minor differences between the two cases, I'd rather not repeat the common stuff in both, following the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). Let me know if you still feel like consolidating the two has any drawbacks.

**Pull request merged**